blacklist/blacklist.go

```diff
@@ -0,0 +1,94 @@
+package blacklist
+
+import (
+	"github.com/Comcast/codex/db"
+	"github.com/Comcast/webpa-common/logging"
+	"github.com/go-kit/kit/log"
+	"sync"
+	"time"
+)
+
+type List interface {
+	InList(ID string) bool
+}
+
+type memList struct {
+	data     map[string]string
+	dataLock sync.RWMutex
+}
+
+func newEmptyMemList() memList {
+	return memList{
+		data: make(map[string]string),
+	}
+}
+
+func (m *memList) InList(ID string) (val string, ok bool) {
+	m.dataLock.RLock()
+	val, ok = m.data[ID]
+	m.dataLock.RUnlock()
+	return
+}
+
+func (m *memList) updateList(data []db.BlackDevice) {
+
+	newData := make(map[string]string)
+	for _, device := range data {
+		newData[device.DeviceID] = device.Reason
+	}
+
+	m.dataLock.Lock()
+	m.data = newData
+	m.dataLock.Unlock()
+}
+
+type dbList struct {
+	logger log.Logger
+
+	listGetter db.RetryListGService
```

Comment:
I think this could be more general; what we're implementing is something that calls a listGetter function and then updates our list.  The listGetter doesn't have to get the list from the database, it just has to have the `GetBlacklist()` function.